vault/cors.go

```diff
@@ -0,0 +1,137 @@
+package vault
+
+import (
+	"errors"
+	"net/http"
+	"regexp"
+	"strings"
+)
+
+var errCORSNotConfigured = errors.New("CORS is not configured")
+
+type CORSConfig struct {
+	enabled          bool
+	allowedOrigins   *[]string
+	allowedHeaders   *map[string]string
+	allowedMethods   *[]string
+	allowCredentials bool
+}
+
+func newCORSConfig() *CORSConfig {
+	defaultOrigins := &[]string{}
+	return &CORSConfig{
+		enabled:        false,
+		allowedOrigins: defaultOrigins,
+		allowedHeaders: &map[string]string{
+			"Access-Control-Allow-Headers": "origin,content-type,cache-control,accept,options,authorization,x-requested-with,x-vault-token",
+			"Access-Control-Max-Age":       "1800",
+			// "Content-Type":                 "text/plain",
+		},
+		allowCredentials: true,
+		allowedMethods: &[]string{
+			http.MethodDelete,
+			http.MethodGet,
+			http.MethodOptions,
+			http.MethodPost,
+			http.MethodPut,
+		},
+	}
+}
+
+func (c *CORSConfig) Enabled() bool {
+	return c.enabled
+}
+
+func (c *CORSConfig) Enable(s string) error {
+	if matched, _ := regexp.MatchString(`\*`, s); matched && len(s) > 1 {
```

Comment:
That's just my normal coding style, and in looking at it again now, it should not be.